sympy/solvers/tests/test_solveset.py

```diff
@@ -757,6 +757,13 @@ def test_solve_trig():
         imageset(Lambda(n, 2*n*pi), S.Integers)
 
 
+def test_issue_13084():
+    n = Dummy('n')
+    f1 = sin(2*x)**2 + 2* sin(2*x) + 1
+    f2 = sin(2*x) + 1
+    assert solveset(f1, x, S.Reals) == ImageSet(Lambda(n, n*pi + 3*pi/4), S.Integers)
+    assert solveset(f1, x, S.Reals) == ImageSet(Lambda(n, n*pi + 3*pi/4), S.Integers)
+
```

Comment:
What about `decompogen`. I figured out that it is not able to handle `pow` type functions.
`2**x`, `2**x + 10`, though there is a check in the function it does not really handle it.